src/shared/components/post/post-listing.tsx

```diff
@@ -1497,6 +1518,20 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
     }
   }
 
+  handleMarkReadPostClick(i: PostListing) {
+    const auth = myAuth();
+    if (auth) {
+      const read =
+        i.props.post_view.read == undefined ? true : !i.props.post_view.read;
+      const form: MarkPostAsRead = {
+        post_id: i.props.post_view.post.id,
+        read,
+        auth,
+      };
+      WebSocketService.Instance.send(wsClient.markPostAsRead(form));
```

Comment:
Was `markPostAsRead` added to the typescript client project? Also, was the work done on the backend to support this?

Reply:
Yes this is already implemented